### DISCUSSION TITLE
collaboration

### DISCUSSION THREAD

**Discussion by martinlarsalbert** [ShipMMG/shipmmg#23] [General]
Hi, my name is Martin.
I'm an industrial PHD at Chalmers in Gothenburg, Sweden, also working at [SSPA](https://www.sspa.se/).
I'm currently researching Parameter Identification of ship manoeuvring models and the MMG model seems to be an interesting candidate for this, which makes this package very interesting. And if I will use it in my research, I can also contribute (I already made a very small PR). 

[https://martinlarsalbert.github.io/blog/](https://martinlarsalbert.github.io/blog/)

**Comment by taiga4112** [comment]
Thank you, Martin.
My name is Taiga MITSUYUKI. Now, I am an Associate professor at Yokohama National University, Japan.

I am not a specialist in naval architect and ship dynamics. But, I have studied a researching project about the system identification and control system of a merchant ship for two years.

I am very happy if this repository is helpful for your project.
If possible, I can support you and collaborate with your research project.

**Reply by martinlarsalbert** [reply]
Very nice!
I'm now in my 2nd year of the PhD,
I have previously looked at roll damping [Prediction-of-roll-motion-using-fully-nonlinear-potential-flow-and-Ikedas-method](https://github.com/martinlarsalbert/Prediction-of-roll-motion-using-fully-nonlinear-potential-flow-and-Ikedas-method). I will now focus more on manoeuvring, so I found your package during my initial research.

I'm happy to help out with this package also, for instance with documentation etc. Just let me know which parts you need help with.

**Reply by taiga4112** [reply]
Thank you.
Of course, I will add my research output to this repository.
However, I do not have an idea for improving this repository from the viewpoint of documentation, etc. because I am a beginner of GitHub.

Do you have any idea?

**Reply by martinlarsalbert** [reply]
I'm not very experienced either, but I think that we could autogenerate a documentation using [https://readthedocs.org/](https://readthedocs.org/).

Perhaps we could create some examples of how this package can be used? (I can try to do that, I will need to do that anyway, in order to understand how it works etc.)

**Reply by taiga4112** [reply]
I tried to make documentation in GitHub pages by using [sphinx](https://www.sphinx-doc.org/en/master/) but not yet completed.

https://shipmmg.github.io/shipmmg/

The branch of this GitHub pages is `gh-pages` which is the submodule of this repository.
I think you can check this repository from `shipmmg/_build/html` directory.
The files in this directory are automatically created by sphinx as the following command.

```sh
$ sphinx-apidoc -f -o docs .
$ cd docs
$ make html
```

In addition, I tried to create an example of how this package can be used as follows.
https://nbviewer.jupyter.org/gist/taiga4112/e53e28053784fef407692becc8226735

I wonder where this explanation should be put in this repository.

**Reply by martinlarsalbert** [reply]
Sorry, I missed your comment here... And I haven't had the time to look at your package for a while, but thanks for mentioning how I can propose changes to the documentation.
By the way, I found this one, which might interest you: https://github.com/cybergalactic/MSS
(There is a python version of this simulator coming up soon, I think)
